packages/api/src/event/queries/get_checklist_events.sql

```diff
@@ -0,0 +1,106 @@
+WITH account_events AS (
+  SELECT event.*
+  FROM
+    event
+  INNER JOIN
+    account
+    ON
+      account.subject::uuid = event.actor_id
+  WHERE
+    account.primaryemail = :email
+)
+
+SELECT
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'archive'
+      AND account_events.action = 'create'
+  ) AS "archiveCreated",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'account'
+      AND account_events.action = 'submit_promo'
+      AND account_events.body ->> 'promoCode' = 'TellYourStory'
```

Comment:
Ah this might be a tough one if there are special promo codes for events, though I like the idea of looking for the first one, rather than any gift code. I do think sometimes we don't use `TellYourStory`, though. @k8lyn6 , would it work to check off the "free storage redeemed" item if they've submitted *any* promo code? Am I right that sometimes you don't use `TellYourStory`?

Reply:
@cecilia-donnelly that's correct -- we use many different codes depending on what avenue they signed up to Permanent through (i.e. Photo Detectives, Photo Managers, etc). It would be better if this was just any code redeemed. Certainly there might be some very rare cases where they redeem a different promo that isn't their original storage, but I think the purpose of this is to prompt them to _learn_ how to redeem a code, and if they've redeemed a different code, I think that's fine (again, it's a very edge case to begin with). Thanks for working on this!